Paint/Paint/MainForm.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            CoordinatesLabel.Text = "";
+        }
+
+        private void SetNewColor(object sender, EventArgs eventArg)
+        {
+            ColorPickerDialog.ShowDialog();
+            currentPenColor = ColorPickerDialog.Color;
+            ColorPickerBtn.BackColor = currentPenColor;
+            drawingPen.Color = currentPenColor;
+        }
+
+        private void DrawByPressedPen(object sender, MouseEventArgs eventArg)
+        {
+            if (isPenDown && lastPoint != Point.Empty)
+            {
+                using (Graphics g = Graphics.FromImage(CanvasPB.Image))
+                {
+                    g.DrawLine(drawingPen, lastPoint, eventArg.Location);
+                }
+                CanvasPB.Invalidate();
+                lastPoint = eventArg.Location;
+            }
+        }
+
+        private void OnMouseDown(object sender, MouseEventArgs eventArg)
+        {
+            isPenDown = true;
+            lastPoint = eventArg.Location;
+        }
+        
+        private void OnMouseUp(object sender, MouseEventArgs eventArg)
+        {
+            isPenDown = false;
+            lastPoint = Point.Empty;
+        }
+
+        private void UpdateImage()
+        {
+            CanvasPB.Image = new Bitmap(lastImage);
+            CanvasPB.Size = new Size(lastImage.Width, lastImage.Height);
+        }
+        
+        private void UpdatePen(object sender, EventArgs e)
+        {
+            drawingPen.Color = currentPenColor;
+            penWidth = (int)PenSizeUD.Value;
+            drawingPen.Width = penWidth;
+        }
+        
+        private Point ShowCustomDialog()
+        {
+            Form prompt = new Form();
+            prompt.Width = 200;
+            prompt.Height = 150;
+            prompt.Text = "New Image";            
+            prompt.AutoSize = false;
+            prompt.MaximizeBox = false;
+            prompt.FormBorderStyle = FormBorderStyle.FixedSingle;
+            var widthLabel = new Label() { Left = 10, Top=20, Text="Width", Width = 60};
+            var widthUD = new NumericUpDown () { Left = 100, Top=20, Width=70, Minimum = 0, Maximum = Decimal.MaxValue};
+            
+            var heightLabel = new Label() { Left = 10, Top=40, Text="Height", Width = 60 };
+            var heightUD = new NumericUpDown () { Left = 100, Top=40, Width=70, Minimum = 0, Maximum = Decimal.MaxValue};
+            
+            var confirmation = new Button() { Text = "Ok", Left=40, Width=100, Top=70 };
+            
+            confirmation.Click += (sender, e) => { prompt.Close(); };
+            
+            prompt.Controls.Add(confirmation);
+            prompt.Controls.Add(widthLabel);
+            prompt.Controls.Add(heightLabel);
+            prompt.Controls.Add(widthUD);
+            prompt.Controls.Add(heightUD);
+            prompt.ShowDialog();
+            return new Point((int)widthUD.Value, (int)heightUD.Value);
```

Comment:
Наконец отрабатываем нажатие кнопки. Если окно диалога было закрыто иным способом, то возвращаем нулевые значения, что просигнализирует вызвавшему методу, что ничего менять не надо.
```suggestion
            if (prompt.ShowDialog() == DialogResult.OK)
                return new Point((int)widthUD.Value, (int)heightUD.Value);
            else
                return new Point(0, 0);
```